gh-clean-branches

```diff
@@ -36,83 +36,75 @@ end=$(printf '\e[0m')
-local_branches=(${local_branches:#* ${default_branch}})        # filter out default_branch
-local_branches=(${local_branches// ##})                        # trim spaces
-
-remote_branches=("${(f)remote_branches_str}")                  # split string by \n to array
-remote_branches=(${remote_branches:#* ${default_branch}})      # filter out default_branch
-remote_branches=(${remote_branches// ##})                      # trim spaces
-
-missing_upstream_branches=(${local_branches:|remote_branches}) # local_branches minus remote_branches
-
-branches_count=${#missing_upstream_branches[@]}
-
-if [[ ${FORCE_DELETE} == true ]]; then
-  delete_flag='-D'
-else
-  delete_flag='-d'
-fi
-
-if [[ ${branches_count} -eq 0 ]]; then
-    printf "%s\n" "${green}No local branches with missing upstream found${end}"
-else
-    printf "%s\n" "${blue}Local branches with missing upstream:${end}"
-    for branch in "${missing_upstream_branches[@]}"; do
-        printf "%s\n" "   ${branch}"
-    done
-
-    if [[ ${DRY_RUN} == false ]]; then
-      [[ ${FORCE_DELETE} == true ]] && printf "%s\n" "${yellow}Force delete is enabled${end}"
+for upstream_name in $(git remote show); do
+  default_branch=$(git remote show ${upstream_name} | awk '/HEAD branch/ {print $NF}')
+
+  printf "%s\n" "${green}Checking out ${default_branch}${end}"
+  git checkout $default_branch
 
+  printf "%s\n" "${green}Pulling ${upstream_name}/${default_branch}${end}"
+  git pull ${upstream_name} ${default_branch}
+  if [[ $? -eq 1 ]]; then
+      printf "%s\n" "${red}Failed to pull, check for uncomitted changes.${end}"
+      exit 1
+  fi
+
+  # list branch locally, and exclude default branch
+  local_branches_str=$(git branch | grep -v "${upstream_name}/${default_branch}")
+  local_branches_str=${local_branches_str/\*?/  } # trim the "*"" on the current branch
+  remote_branches_str=$(git branch -r | grep ${upstream_name} | grep -v "${upstream_name}/${default_branch}") # exclude default branch
+  remote_branches_str=${remote_branches_str//${upstream_name}\// } # trim the "upstream_name/" from branch names
+
+  if [[ ${VERBOSE} == true ]]; then
+    printf "%s\n%s\n" "${blue}Local branches:${end}" "${local_branches_str}"
+    printf "%s\n%s\n" "${blue}Remote branches:${end}" "${remote_branches_str}"
+  fi
+
+  setopt extended_glob
+  local_branches=("${(f)local_branches_str}")                    # split string by \n to array
+  local_branches=(${local_branches:#* ${default_branch}})        # filter out default_branch
+  local_branches=(${local_branches// ##})                        # trim spaces
+  remote_branches=("${(f)remote_branches_str}")                  # split string by \n to array
+  remote_branches=(${remote_branches:#* ${default_branch}})      # filter out default_branch
+  remote_branches=(${remote_branches// ##})                      # trim spaces
+
+  missing_upstream_branches=(${local_branches:|remote_branches}) # local_branches minus remote_branches
```

Comment:
`missing_upstream_branches` will find and delete a local branch that has an upstream in another remote.

Example: 
```bash
⇒  git remote show             
origin-1
origin-2
```

```bash
⇒  git branch -a | cat
* main
  test-branch
  remotes/origin-2/main
  remotes/origin-2/test-branch
  remotes/origin-1/HEAD -> origin/master
  remotes/origin-1/add-module
  remotes/origin-1/main
  remotes/origin-1/master
```
I have two origins, `origin-1` and `origin-2`. 
The local branch `test-branch` has upstream on `origin-2` but doesn't exist on `origin-1`.

When iterating over the upstreams having `upstream_name` as `origin-1`, the branch `test-branch` will be added to `missing_upstream_branches` as it is not included in `remote_branches`. The branch `test-branch` will be deleted in line 79, although it has an upstream in `origin-2`.